backend/src/main/resources/application.yaml

```diff
@@ -8,8 +8,10 @@ spring:
     password: api123
     url: jdbc:postgresql://localhost:${SR_DB_PORT:5432}/simple_report
     hikari:
-      maximum-pool-size: 20
-      max-lifetime: 900000 # Maximum lifetime for a connection to be retained in the pool, in milliseconds, once it is closed.
+      maximum-pool-size: 50 # Maximum size that the pool is allowed to reach, including both idle and active connections. Effectively reflects number of concurrent database request threads. (Ex: A size of 50 represents a maximum capacity of 50 concurrent DB requests.)
+      max-lifetime: 600000 # Maximum lifetime for a connection to be retained in the pool, in milliseconds, once it is closed.
+      minimum-idle: 30 # Minimum number of idle connections to be retained in the pool. When a connection has been idle for the length of time specified in `idle-timeout`, it will be purged from the pool. This will allow for any un-closed, inactive transactions to be cleaned up more readily.
```

Comment:
The docs strongly suggest that we should not set this. What's the expected outcome of setting this vs. letting Hikari manage a fixed-size pool?

https://github.com/brettwooldridge/HikariCP#gear-configuration-knobs-baby

Reply:
I did see that...I debated for some time whether to change this or not. The docs indicate that Hikari should be set as a fixed-size connection pool for best performance, and to compensate for demand spikes...but, that would be assuming that the application is properly designed and optimized. My thought is that there is a potential we have a design quirk somewhere that is causing connections to be held onto for far longer than is necessary for a query to complete. By instituting this setting, I'm expecting Hikari to forcibly reclaim an idle connection, which should take care of that stagnation. Without this setting, I think Hikari will just let the connections languish at idle if they haven't been properly closed by our code.

Setting `minimum-idle` to 30 raises our constant connection number by 10, which is a 50% increase over what we have now. Increasing that `maximum-pool-size` allows us to engage 20 additional connections when they are needed.

Without further investigation into the application and its behaviors, I cannot confirm for sure what level of benefit this will have, if any. It's purely a hypothesis. That said, anecdotes from other developers indicate that introducing `minimum-idle` has little, if any, performance impact.

I'm more than happy to wind this back if the team disapproves. I think this is going to require a holistic look at the application, instead of just these few settings, and these values will by no means be final.